python/ray/tests/test_actor_advanced.py

```diff
@@ -1331,6 +1331,87 @@ def ready(self):
     #     ray.get_actor("actor_2", namespace="namespace")
 
 
+def test_get_actor_race_condition(shutdown_only):
+    @ray.remote
+    class Actor:
+        def ping(self):
+            return "ok"
+
+    @ray.remote
+    def getter(name):
+        try:
+            try:
+                actor = ray.get_actor(name)
+            except Exception:
+                print("Get failed, trying to create", name)
+                actor = Actor.options(name=name).remote()
+        except Exception:
+            print("Someone else created it, trying to get", name)
+            actor = ray.get_actor(name)
+        return ray.get(actor.ping.remote())
+
+    def do_run(name, concurrency=4):
+        name = "actor_" + str(name)
+        tasks = [getter.remote(name) for _ in range(concurrency)]
+        result = ray.get(tasks)
+        try:
+            ray.kill(ray.get_actor(name))  # Cleanup
+        except Exception:
+            pass
+        return result
+
+    for i in range(50):
+        # TODO(sang): Currently increasing concurrency
+        # to 8 will make test flaky due to this issue
+        # https://github.com/ray-project/ray/issues/20129
+        CONCURRENCY = 4
+        results = do_run(i, concurrency=CONCURRENCY)
```

Comment:
Why does this bug affect this repro, don't we ray.kill only for cleanup here?

Reply:
I think in the next round (next iteration), get_actor returns the killed actor instead of raising an ValueError.

Reply:
(So it seems to take some time between ray.kill and ray.get_actor reflects that. IIUC that shouldn’t happen, but it does, so there must be a bug there)

Reply:
sounds like this test will be flakey.

Reply:
Yeah good point! Let me add a safe check and increase the concurrency here

Reply:
actually, this was the typical ownership issue. It passes consistently after I changed the lifetime to detached. (So, actor was killed because the idle worker owns the actor handle, which is subsequently killed by the worker pool)